sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -436,14 +436,19 @@ class AstBuilder extends DataTypeAstBuilder
         val (relationCtx, options, cols, partition, ifPartitionNotExists, byName)
         = visitInsertIntoTable(table)
         withIdentClause(relationCtx, ident => {
-          InsertIntoStatement(
+          val insertIntoStatement = InsertIntoStatement(
             createUnresolvedRelation(relationCtx, ident, options),
             partition,
             cols,
             query,
             overwrite = false,
             ifPartitionNotExists,
             byName)
+          if (conf.getConf(SQLConf.OPTIMIZE_INSERT_INTO_VALUES_PARSER)) {
+            EvaluateUnresolvedInlineTable.evaluate(insertIntoStatement)
```

Comment:
looking at it again, I guess we can just call `ResolveInlineTables.apply`? Then we don't need to create `EvaluateUnresolvedInlineTable`?

Reply:
We brainstormed about this earlier, but realized that doing so would be invoking an analyzer rule (starting with `Resolve`, and inheriting from the Catalyst rule classes) directly from the parser. With database management system implementation, it's a best practice to separate concerns between the different distinct phases of SQL compilation, including parsing, analysis, logical optimization, logical to physical planning, and execution binding.

Thus the refactor to call `EvaluateUnresolvedInlineTable.evaluate` instead. The code does the same steps, it's just now a helper function instead of calling a Catalyst rule directly from the parser. This is mostly a style thing.